memcache/memcache.go

```diff
@@ -171,6 +180,80 @@ type Item struct {
 	casid uint64
 }
 
+// MetaGetFlags are the flags supported by MetaGet function
+type MetaGetFlags struct {
+
+	//Should be true if the key passed to MetaGet is in base64Format.
+	//Same as Meta Get -b option
+	IsKeyBase64 bool
+
+	//Set this to true if cas token should be included in meta get response metadata
+	//Same as Meta Get -c option
+	ReturnCasTokenInResponse bool
+
+	//Set this to true if client flags should be included in meta get response metadata
+	//Same as Meta Get -f option
+	ReturnClientFlagsInResponse bool
+
+	//Set this to true if you want to know if an item is hit before
+	//Same as Meta Get -h option
+	ReturnItemHitInResponse bool
+
+	//Set this to true if key should be included in meta get response metadata
+	//Same as Meta Get -k option
+	ReturnKeyInResponse bool
+
+	//Set this to true if last time since item was accessed in Seconds
+	//be included in meta get response metadata
+	//Same as Meta Get -l option
+	ReturnLastAccessedTimeSecondsInResponse bool
+
+	//Set this to true if item size in bytes should be included in meta get response metadata
+	//Same as Meta Get -s option
+	ReturnItemSizeBytesInResponse bool
+
+	//Set this to true if remaining ttl of the item should be included in meta get response metadata
+	//Same as Meta Get -t option
+	ReturnTTLRemainingSecondsInResponse bool
+
+	//Set this to true to access an item without causing it to be "bumped" to the head
+	//of the LRU. This also avoids marking an item as being hit or updating its last
+	//access time.
+	DontBumpItemInLRU bool
+
+	//opaque value, consumes a token and copies back with response
+	//Same as Meta Get -O option
+	OpaqueToken *string
+
+	//If supplied, and metaget does not find the item in cache, it will
+	//create a stub item with the key and TTL as supplied.
+	//todo update this
+	//Same as Meta Get -N option
+	VivifyTTLToken *int32
+
+	//todo add comment
+	ReCacheTTLToken *int32
+
+	//todo add comment
+	UpdateTTLToken *int32
+}
+
+//MetaResponseMetadata contains response metadata from meta commands
+//Some of these values can be nil if the corresponding meta flag is not set while calling
+//meta functions
+type MetaResponseMetadata struct {
+	//todo add comments
+	CasId                 *uint64
+	TTLRemainingInSeconds *int32
+}
+
+//MetaGetResponse holds the response of MetaGet command
+type MetaGetResponse struct {
```

Comment:
Wouldn't all meta command responses contain a value and response meta data? We might be able to consolidate these into one response object.

Reply:
Not all commands return value. Only Get and Arithmetic return values. I want to include value field only where the command returns value.

Reply:
Yes. If they do not include any response flags then we will return a nil response metadata.